doc/arkode/guide/source/Usage/LSRKStep/User_supplied.rst

```diff
@@ -0,0 +1,49 @@
+.. ----------------------------------------------------------------
+   Programmer(s): Daniel R. Reynolds @ SMU
+                  David J. Gardner @ LLNL
+   ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2024, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _LSRKSTEP.Usage.UserSupplied:
+
+User-supplied functions
+=============================
+
+Currently, there is only one user-supplied function for LSRKStep (in addition to the :c:type:`ARKRhsFn` arguments that define the IVP).
+
+
+
+
+.. _LSRKStep.Usage.dom_eig:
+
+The dominant eigenvalue estimation
+----------------------------------
+
+When running LSRKStep with either the RKC or RKL methods, the user must supply a dominant eigenvalue estimation function of type :c:type:`ARKDomEigFn`:
+
+.. c:type:: int (*ARKDomEigFn)(sunrealtype t, N_Vector y, N_Vector fn, sunrealtype* lambdaR, sunrealtype* lambdaI, void* user_data, N_Vector temp1, N_Vector temp2, N_Vector temp3);
```

Comment:
@david -- this directly conflicts with @Steven-Roberts request that these be added.

I personally like Steven's idea of including them, since the dominant eigenvalue obviously relates to the Jacobian, and this is the list of inputs we provide to users for the corresponding user-supplied routines.

Reply:
I don't think we should pass the temp vectors to the Jacobian function either but that function signature has been around for a long time. I'd prefer that the function signature include only quantities needed to compute the outputs and not extra workspace vectors as well.